fractal_funcs.py

```diff
@@ -0,0 +1,191 @@
+"""
+fractal_funcs.py
+
+This module contains functions for generating various
+fractal patterns designed for use with a Tkinter canvas.
+The fractal generating functions utilize recursion to
+create complex and visually appealing structures.
+
+Functions:
+    fractal_canopy()
+        Recursively draws a fractal canopy (a tree-like structure) on the provided Tkinter canvas.
+"""
+
+import tkinter as tk
+import numpy as np
+from numpy import radians as rad
+from helper_funcs import draw_sine_wave_segment
+
+
+def fractal_canopy(
+    canvas: tk.Canvas,
+    x: float,
+    y: float,
+    off_angle: float = 0,
+    angle_delta: float = 10,
+    start_angle: float = -90,
+    n_splits: int = 2,
+    n_iters: int = 3,
+    length_ratio: float = 0.75,
+    init_length: float = 200,
+    wave_amp: float = 0,
+    width: float = 1,
+    width_ratio: float = 0.75,
+    color="#000000",
```

Comment:
we should stay consistent and add type hint to the color as well. Furthermore we should put restrictions on how many elements the gradient list has as I noticed it impacts performance when generating the tree and creates errors too. Here for example it just stretches the range if the range contains too many values: 
![Screenshot 2024-11-29 193950](https://github.com/user-attachments/assets/f88c7fcf-8676-427e-9537-2ba5eb1794f8)

Reply:
Team has decided the PR can be merged in the current state. Resolving the comment.